stablehlo/dialect/StablehloOps.cpp

```diff
@@ -3952,17 +3953,19 @@ ParseResult ReduceOp::parse(OpAsmParser& parser, OperationState& result) {
   return success();
 }
 
-LogicalResult ReduceOp::verify() {
-  // Check that there are even number of operands and >= 2.
-  if (getNumOperands() % 2 != 0 || getOperands().empty())
-    return emitOpError() << "expects the size of operands to be even and >= 2";
-
+LogicalResult ReduceOp::inferReturnTypeComponents(
+    MLIRContext *, Optional<Location> location, ValueShapeRange operands,
+    DictionaryAttr attributes, RegionRange regions,
+    SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes) {
   // Collect the input and init-value operands. Note that the operand-type is
   // enforced as "TensorType" by ODS.
-  int64_t numInputs = getNumOperands() / 2;
+  if (operands.size() % 2 != 0 || operands.size() == 0)
+    return emitOptionalError(location, "expects the size of operands to be even and >= 2");
+  int64_t numInputs = operands.size() / 2;
   auto operandTensorTypes = llvm::to_vector<4>(llvm::map_range(
-      getOperandTypes(),
-      [](Type t) -> TensorType { return t.cast<TensorType>(); }));
+      operands.getTypes(),
+      [](Type t) -> TensorType
+      { return t.cast<TensorType>(); }));
```

Comment:
Probably shouldn't change formatting here? In the near future when #49 lands, we'll have `.clang-format` + we'll have CI complaining if formatting isn't compliant with `clang-format`. In the meanwhile, this part can be fixed manually.

Reply:
Sure, I tried this in my `settings.json` MS Code:
```
"clang-format.style": "{BasedOnStyle: google}",
// "C_Cpp.formatting": "Disabled"
// For vscode >= 1.33, try this instead of "C_Cpp.formatting": "Disabled"
"[cpp]": {
    "editor.defaultFormatter": "xaver.clang-format"
},
```
After auto-format, the code looks exactly you suggested.